server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -0,0 +1,136 @@
+            } catch (IndexOutOfBoundsException | IllegalArgumentException e) {
+                throw new IllegalArgumentException(restAction + " does not begin with a valid REST method");
+            }
+            logger.info("Registering: " + method + " " + uri);
+            restActionsAsRoutes.add(new Route(method, uri));
+        }
+        this.routes = unmodifiableList(restActionsAsRoutes);
+        this.discoveryExtension = discoveryExtension;
+        this.transportService = transportService;
+    }
+
+    @Override
+    public String getName() {
+        return "send_to_extension_action";
+    }
+
+    @Override
+    public List<Route> routes() {
+        return this.routes;
+    }
+
+    @Override
+    public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+        Method method = request.getHttpRequest().method();
+        String uri = request.getHttpRequest().uri();
+        if (uri.startsWith(uriPrefix)) {
+            uri = uri.substring(uriPrefix.length());
+        }
+        String message = "Forwarding the request " + method + " " + uri + " to " + discoveryExtension;
+        logger.info(message);
+        try {
+            // Notify user the request was acted on.
+            return channel -> channel.sendResponse(new BytesRestResponse(RestStatus.ACCEPTED, message));
+        } finally {
+            final TransportResponseHandler<RestExecuteOnExtensionResponse> restExecuteOnExtensionResponseHandler =
+                new TransportResponseHandler<RestExecuteOnExtensionResponse>() {
+
+                    @Override
+                    public RestExecuteOnExtensionResponse read(StreamInput in) throws IOException {
+                        return new RestExecuteOnExtensionResponse(in);
+                    }
+
+                    @Override
+                    public void handleResponse(RestExecuteOnExtensionResponse response) {
+                        logger.info("Received response from extension: {}", response.getResponse());
+                    }
+
+                    @Override
+                    public void handleException(TransportException exp) {
+                        logger.debug(new ParameterizedMessage("Extension initialization failed"), exp);
+                    }
+
+                    @Override
+                    public String executor() {
+                        return ThreadPool.Names.GENERIC;
+                    }
+                };
+            try {
+                transportService.sendRequest(
+                    discoveryExtension,
+                    ExtensionsOrchestrator.REQUEST_REST_EXECUTE_ON_EXTENSION_ACTION,
+                    new RestExecuteOnExtensionRequest(method, uri),
+                    restExecuteOnExtensionResponseHandler
+                );
+            } catch (Exception e) {
+                logger.info("Failed to send Register REST Actions request to OpenSearch", e);
```

Comment:
Correct me if wrong here. The log message should be?
```suggestion
                logger.info("Failed to send Register REST Actions request to Extension", e);
```

Reply:
You're not wrong. Started with a copypaste from ExtensionsRunner and forgot to change it...

Reply:
Is this remaining?

Reply:
I had the old copy of code. Ignore the previous comment.